src/plans/nonlinear_least_squares_plan.jl

```diff
@@ -99,10 +99,13 @@ function get_gradient!(
 end
 
 function get_gradient!(
-    M::AbstractManifold, X, nlso::NonlinearLeastSquaresObjective{InplaceEvaluation}, p
+    M::AbstractManifold, 
+    X,
+    nlso::NonlinearLeastSquaresObjective{InplaceEvaluation},
+    p,
+    Jval = zeros(nlso.num_components, manifold_dimension(M))
 )
     basis_p = _maybe_get_basis(M, p, nlso.jacobian_tangent_basis)
-    Jval = zeros(nlso.num_components, manifold_dimension(M))
```

Comment:
This is where all the allocations come from, in my case, I give a sparse jacobian as input.